common/browser_context_options.go

```diff
@@ -132,3 +134,48 @@ func (b *BrowserContextOptions) Parse(ctx context.Context, opts goja.Value) erro
 	}
 	return nil
 }
+
+// WaitForEventOptions are the options used by
+// the browserContext.waitForEvent API.
+type WaitForEventOptions struct {
+	Timeout     time.Duration
+	PredicateFn goja.Callable
+}
+
+// NewWaitForEventOptions created a new instance of
+// WaitForEventOptions with a default timeout.
+func NewWaitForEventOptions(defaultTimeout time.Duration) *WaitForEventOptions {
+	return &WaitForEventOptions{
+		Timeout: defaultTimeout,
+	}
+}
+
+// Parse will parse the options or a callable predicate function.
+// It can parse only a callable predicate function or an object
+// which contains a callable predicate function and a timeout.
+func (w *WaitForEventOptions) Parse(ctx context.Context, optsOrPredicate goja.Value) error {
+	var isCallable bool
+	if gojaValueExists(optsOrPredicate) {
```

Comment:
Early return?

```go
	if !gojaValueExists(optsOrPredicate) {
		return nil
	}
	// actual code is here
```

Reply:
👍 resolved in 3d1f55840554dd1fda8ce87161e1e5092b2ac8b6.

Reply:
Thanks!